### DISCUSSION TITLE
compile error when I use json in template

### DISCUSSION THREAD

**Discussion by erickksun** [nlohmann/json#2977] [Q&A] [answered]
Hi,
my code:
```
template<typename T>
void foo(const T &config)
{
    config["some_key"].get<std::string>();
}
```
the compile error:
```
error: expected primary-expression before ‘>’ token
     config["some_key"].get<std::string>();
error: expected primary-expression before ‘)’ token
     config["some_key"].get<std::string>();
```
_______________________________________________________
why ???

**Comment by erickksun** [comment]
$ arm-linux-gnueabihf-gcc --version
arm-linux-gnueabihf-gcc (GNU Toolchain for the A-profile Architecture 8.3-2019.03 (arm-rel-8.36)) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**Comment by nlohmann** [comment, marked as answer]
With Xcode, I get the following message:

```
Use 'template' keyword to treat 'get' as a dependent template name
```

This should work:

```cpp
template<typename T>
void foo(const T &config)
{
    config["some_key"].template get<std::string>();
}
```

**Reply by erickksun** [reply]
> With Xcode, I get the following message:
> 
> ```
> Use 'template' keyword to treat 'get' as a dependent template name
> ```
> 
> This should work:
> 
> ```c++
> template<typename T>
> void foo(const T &config)
> {
>     config["some_key"].template get<std::string>();
> }
> ```

thanks a lot.
